DAOS-13268 cart: Fix UCX to use rdmacm.

jgmoore-or opened the pull request:
Test-provider: ucx+dc_x

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

jgmoore-or committed:
DAOS-13268 cart: Fix UCX to use rdmacm.

Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/2/execution/node/674/log

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/2/execution/node/814/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/2/execution/node/721/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented on src/cart/crt_init.c:
```diff
@@ -307,6 +307,11 @@ static int data_init(int server, crt_init_options_t *opt)
+	setenv("UCX_SOCKADDR_TLS_PRIORITY", "rdmacm", 1);
+	setenv("HG_LOG_LEVEL", "warning", 1);
+	setenv("HG_LOG_SUBSYS", "na", 1);
+	//setenv("D_LOG_STDERR_IN_LOG", "1", 1);
```
(style)  do not use C99 // comments

daosbuild1 replied:
```suggestion
	/* setenv("D_LOG_STDERR_IN_LOG", "1", 1); */
```

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12075/3/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/3/execution/node/145/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/3/execution/node/289/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/3/execution/node/292/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented on src/cart/crt_init.c:
```diff
@@ -307,6 +307,12 @@ static int data_init(int server, crt_init_options_t *opt)
 	if (server)
 		setenv("UCX_IB_FORK_INIT", "n", 1);
 
+	//setenv("UCX_SOCKADDR_TLS_PRIORITY", "rdmacm", 1);
```
(style)  do not use C99 // comments

daosbuild1 replied:
```suggestion
	/* setenv("UCX_SOCKADDR_TLS_PRIORITY", "rdmacm", 1); */
```

daosbuild1 replied:
(style)  do not use C99 // comments

daosbuild1 replied:
```suggestion
	/* setenv("UCX_SOCKADDR_TLS_PRIORITY", "rdmacm", 1); */
```

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12075/4/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/4/execution/node/119/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/4/execution/node/289/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/4/execution/node/292/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12075/5/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/5/execution/node/144/log

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/5/execution/node/719/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/5/execution/node/672/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-verbs-provider: true
Skip-func-hw-test-medium-ucx-provider: false
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/6/execution/node/765/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-verbs-provider: true
Skip-func-hw-test-medium-ucx-provider: false
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/7/execution/node/719/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/7/execution/node/672/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Scan Leap 15.4 RPMs completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/8/execution/node/523/log

jgmoore-or committed:
Merge branch 'master' into jgm/daos-13268

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Scan Leap 15.4 RPMs completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/10/execution/node/524/log

jgmoore-or committed:
Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-scan-leap15-rpms: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/14/execution/node/658/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/15/execution/node/642/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/16/execution/node/648/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/17/execution/node/611/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12075/18/testReport/(root)/

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/19/execution/node/611/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/20/execution/node/611/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/21/execution/node/642/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/22/execution/node/611/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12075/23/testReport/(root)/

jgmoore-or committed:
Merge branch 'master' into jgm/daos-13268

Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-scan-leap15-rpms: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x
PR-repos: mercury@PR-100:5

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/24/execution/node/699/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/25/execution/node/660/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/26/execution/node/660/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/27/execution/node/660/log

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/27/execution/node/753/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build on Leap 15.4 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/28/execution/node/294/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/28/execution/node/333/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/28/execution/node/290/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/29/execution/node/660/log

jgmoore-or committed:
DAOS-13268 cart: Fix UCX.

Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/30/execution/node/768/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12075/30/execution/node/721/log

jgmoore-or committed:
DAOS-13268 cart: Upgrade to UCX 1.14.

Skip-build-leap15-icc: true
Skip-build-el8-gcc: true
Skip-build-ubuntu20-rpm: true
Skip-build-leap15-rpm: true
Skip-func-test-vm: true
Skip-unit-tests: true
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-medium-verbs-provider: true
Test-provider: ucx+dc_x
PR-repos: openucx@PR-9:2

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
